lib/core.js

```diff
@@ -841,54 +495,72 @@ module.exports = class Autocore extends ReadyResource {
     await core.ready()
 
     // clone state from previous core
-    const batch = await this._migrateCurrentCore(core, length, indexers)
-
-    const old = this.core
-    const oldOriginal = this.originalCore
+    await this._migrateCurrentCore(core, length, indexers)
 
+    const closing = this.originalCore.close()
     this.originalCore = core
-    this.core = batch
 
-    await old.close()
-    await oldOriginal.close()
+    await closing
 
     if (this._isSystem()) this._registerSystemCore()
 
-    for (const session of this.sessions) {
-      if (!session.snapshotted) session.emit('migrate')
-    }
+    // for (const session of this.sessions) {
+    //   if (!session.snapshotted) session.emit('migrate')
+    // }
```

Comment:
we need the migration event in keet actually, but we should just emit this on the base itself vs on the view when all views have migrated, perfectly fine

Reply:
yeh either on the base or on the core during moveTo